### PR TITLE
Add Windows support

### PR DESCRIPTION
**Pull request by dtip**
The basic tests are now passing on Windows!

There's a huge number of commits here. Commits to do with porting are prepended with 'Windows:'. The rest are either to do with CI or they're tidying up the code. There was a lot of old commented-out code which has now been removed and I've fixed some indentation and other whitespace issues.

### CODE REVIEW

**Comment by FussyDuck** [comment]
[CLA assistant check](http://claassistant.ecmwf.int:5000/ecmwf/magics?pullRequest=3) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](http://claassistant.ecmwf.int:5000/ecmwf/magics?pullRequest=3) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: StephanSiemen<br/>:x: dtip<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](http://claassistant.ecmwf.int:5000/check/ecmwf/magics?pullRequest=3) it.</sub>

File: src/basic/LayoutManager.cc [hunk 1]
```diff
@@ -138,9 +138,9 @@ BasicSceneNode* MagMLLayoutManager::operator()(BasicSceneNode* parent, BasicPosi
 	 DisplayType display = layout.display();
 	 
 	 if ( actions_.empty() ) {
-		 	actions_[INLINE] = &MagMLLayoutManager::inline_display;
-		 	actions_[ABSOLUTE] = &MagMLLayoutManager::absolute;
-		 	actions_[BLOCK] = &MagMLLayoutManager::block;
+			actions_[M_DT_INLINE] = &MagMLLayoutManager::inline_display;
```
Comment (sylvielamythepaut) [inline src/basic/LayoutManager.cc]: I wsa wandering why you used this convention M_DT ? what is the DT standing for ?

File: src/common/MatrixHandler.h [hunk 2]
```diff
@@ -146,12 +146,12 @@ public :
 
 				double min = *std::min_element(distances.begin(), distances.end());
 
-                map<double, pair< std::pair<double, double>,  pair<int, int> > >::iterator near = helper.find(min);
```
Comment (sylvielamythepaut) [inline src/common/MatrixHandler.h]: Why did you change the variable : just for readability or there is something more we need to know ?

File: src/common/PolarStereographicProjection.cc [hunk 3]
```diff
@@ -428,7 +428,7 @@ void PolarStereographicProjection::gridLongitudes(const GridPlotting& grid)  con
 	{
 		if (*lon < xmin_ ) continue;
 		if (*lon > xmax_ ) continue;
-		Polyline poly;
+		magics::Polyline poly;
```
Comment (sylvielamythepaut) [inline src/common/PolarStereographicProjection.cc]: Same thing here, why did you have to specify the namespace ?

File: test/contour.c [hunk 4]
```diff
@@ -11,15 +11,16 @@
 #include "magics_api.h"
 #include <stdio.h>
 
+#define NUM_FORMATS 3
```
Comment (sylvielamythepaut) [inline test/contour.c]: Why did you have to do this one ?

File: src/common/GeoRectangularProjection.cc [hunk 5]
```diff
@@ -253,7 +253,7 @@ void GeoRectangularProjection::gridLatitudes(const GridPlotting& grid)  const
 	{
 		if ( *lat < -90 ) continue;
 		if ( *lat > 90 ) continue;
-		Polyline poly;
+		magics::Polyline poly;
```
[the remark this answers is not in the document]
Comment (dtip) [reply src/common/GeoRectangularProjection.cc]: It's because win_time.h (our sys/time.h replacement) includes windows.h, which defines a load of things. One of them is a Polyline function, which then clashes with the Magics Polyline function.

In terms of this specific file, it's because we include MatrixHandler.h which includes Timer.h which includes win_time.h.

File: src/common/MatrixHandler.h (hunk 2 above)
Comment (dtip) [reply src/common/MatrixHandler.h]: It's because windef.h (which is included by windows.h, which we include in win_time.h) defines macros for 'near' and 'far' so we can't use those as variable names. I've left the info in the [commit](https://github.com/ecmwf/magics/pull/3/commits/5e26938f83d1281c184abf6467aeb16917d35e3a) message for future reference.

File: src/basic/LayoutManager.cc (hunk 1 above)
Comment (sylvielamythepaut) [reply src/basic/LayoutManager.cc]: Ah , OK it is not for dtip ;).. Makes also sense now !

File: src/common/PolarStereographicProjection.cc (hunk 3 above)
Comment (dtip) [reply src/common/PolarStereographicProjection.cc]: Same as the others - because of the clash with the 'Polyline' function defined in windows.h

Comment (sylvielamythepaut) [reply src/common/PolarStereographicProjection.cc]: I have my answer now

File: src/common/MatrixHandler.h (hunk 2 above)
Comment (sylvielamythepaut) [reply src/common/MatrixHandler.h]: Sorry, I had not read the commit message .. 
Good to know too ..

File: test/contour.c (hunk 4 above)
Comment (dtip) [reply test/contour.c]: Well apparently MSVC doesn't support C99 - which means that it doesn't consider `const int num_formats = 3` to actually be a constant. It seems a bit mad really. Then, if `num_formats` isn't constant, the code won't compile, because MSVC also doesn't support variable length arrays. See [this commit](https://github.com/ecmwf/magics/pull/3/commits/26d1622946b86b8542366b925a5992a6ec016129) for an example fix.

We could have fixed `contour.c` in a similar way, but defining a macro seemed slightly more straightforward!

Comment (sylvielamythepaut) [reply test/contour.c]: That's what we thought ! and I had missed the change in MagicsCalls.cc

**Comment by dtip** [comment]
Almost!
